cryptoconditions/crypto.py

```diff
@@ -1,269 +1,161 @@
+            signature (bytes): {base64|base32|base16|hex|bytes} signature to be verified
+            encoding(str): {'bytes'|'hex'|'base16'|'base32'|'base58'|'base64'}. Encoding of the signature. 
+                           Defaults to 'base58'.
         """
+
+        # The reason for using raw_signatures here is because the verify method of pynacl expects the message
+        # and the signature to have the same encoding. Basically pynacl does:
+        #   encoder.decode(signature + message)
+        raw_signature = _get_nacl_encoder(encoding).decode(signature)
         try:
-            if not isinstance(data, bytes):
-                data = data.encode('ascii')
-            if encoding == 'base58':
-                super().verify(base58.b58decode(signature), data, prefix=prefix)
-            else:
-                super().verify(signature, data, prefix=prefix, encoding=encoding)
-        except ed25519.BadSignatureError:
+            super().verify(data, raw_signature)
+        except nacl.exceptions.BadSignatureError:
             return False
 
         return True
 
-    def to_ascii(self, prefix="", encoding='base58'):
+    def encode(self, encoding='base58'):
         """
-        convert external value to ascii with specified encoding
+        Encode the public key
 
         Args:
-            prefix (str):
-            encoding (str): {'base58'|'base64'|'base32'|'base16'|'hex'}
+            encoding(str): {'bytes'|'hex'|'base16'|'base32'|'base58'|'base64'}. Encoding in which the public
+                           key should be returned. Defaults to 'base58'.
 
         Returns:
-            bytes: encoded string
+            The public key encoded with `encoding`.
         """
-        if encoding == 'base58':
-            return base58.b58encode(self.vk_s).encode('ascii').decode('ascii').rstrip("=").encode()
-        else:
-            return super().to_ascii(prefix=prefix, encoding=encoding)
-
-    @staticmethod
-    def encode(public_base64):
-        """
-        Encode the public key represented by base64 to base58
-
-        Args:
-            public_base64
-        """
-        return Ed25519SigningKey.encode(public_base64)
-
-    @staticmethod
-    def decode(public_base58):
-        """
-        Decode the base58 public_value to base64
-
-        Args:
-            public_base58
-        """
-        return Ed25519SigningKey.decode(public_base58)
+        return super().encode(encoder=_get_nacl_encoder(encoding))
 
 
 def ed25519_generate_key_pair():
     """
-    Generate a new key pair and return the pair encoded in base58
+    Generate a new key pair.
+
+    Returns:
+        A tuple of (private_key, public_key) encoded in base58.
     """
-    sk, vk = ed25519.create_keypair()
+    sk = Ed25519SigningKey.generate()
     # Private key
-    private_value_base58 = Ed25519SigningKey(base58.b58encode(sk.to_bytes())).to_ascii()
+    private_value_base58 = sk.encode(encoding='base58')
```

Comment:
rename to `signing_key_base58` or `sk_base58`
or better: add `encoding` as a `kwarg`: `ed25519_generate_key_pair(encoding='base58')`